src/web/components/elements/UnsafeEmbedBlockComponent.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect, useRef } from 'react';
+import { css } from 'emotion';
+
+type Props = {
+    html: string;
+    alt: string;
+};
+
+let numberOfIntervals = 0;
+export const UnsafeEmbedBlockComponent = ({ html, alt }: Props) => {
+    const [iframeHeight, setIframeHeight] = useState<number>(0);
+    const iFrameRef = useRef<HTMLIFrameElement>(null);
+
+    useEffect(() => {
+        if (numberOfIntervals >= 12) return;
+
+        const timer = setInterval(() => {
+            const doc =
+                iFrameRef.current &&
+                iFrameRef.current.contentWindow &&
+                iFrameRef.current.contentWindow.document;
+
+            numberOfIntervals += 1;
+
```

Comment:
I wonder if we could align the DCR resizing contract here. We already have resizing for atoms using iframeMessenger and postMessage with a listener on DCR. We could remove this different way of doing it and align on one approach for iframes as you're building the iframe yourself?

Then we could use this https://github.com/guardian/dotcom-rendering/blob/55ccdbe592a5895133b67234ea18fb81f1467e0c/src/web/browser/atomIframe/init.ts#L10 and extend it out from just `atom-iframe`, to fire `set-height` from the iframe.

You'd have to add it in the inner iframe using the `<gu-script>` mechanism as there's three steps to it:

- Page loads, all iframes on the page are rendered
- DCR starts listening for postmessage set-height from recognised iframes (Currently atom-iframe but I think we should extend that)
- DCR runs over all recognised iframes to convert `<gu-script>` to `script` tags, those contain postMessage `set-height` messages

You would need to implement something that `iframeMessenger` within the iframe https://github.com/guardian/interactive-us-campaign-finance-charts/blob/f746ae26e1de87c6c1f2f2bfa58df5e37018a34d/rich-zips-table/src/js/iframeMessenger.js

I guess best thing to do would be to inline that script in the iframe?

You could then avoid the hydration I think because you're using the existing mechanism and avoiding multiple ways for setting heights on iframes

Reply:
> You would need to implement something that iframeMessenger within the iframe
I assumed that the HTML code came from frontend, so would we need to add `iframeMessenger.js` within the frontend returned HTML?